File: 0045 Jump Game II/0045.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int jumps = 0;
        int cur = 0;
        int reach = 0;
        
        for (int i = 0; i < nums.size() - 1; i++){
            reach = max(reach, i + nums[i]);
            if (i == cur){
                jumps++;
                cur = reach;
            }
        }
        
        return jumps;
    }
};

File: 0045 Jump Game II/0045.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        jumps, cur, reach = 0, 0, 0
        
        for i in range(len(nums) - 1):
            reach = max(reach, i + nums[i])
            if i == cur:
                jumps += 1
                cur = reach
        
        return jumps
